### PR TITLE
Add direct access to auxiliary fields for forcing functions

### PR DESCRIPTION
**Pull request by jagoosw**
To implement #2730

### CODE REVIEW

File: src/Models/NonhydrostaticModels/calculate_nonhydrostatic_tendencies.jl [hunk 1]
```diff
@@ -49,6 +49,7 @@ function calculate_interior_tendency_contributions!(model)
     background_fields    = model.background_fields
     velocities           = model.velocities
     tracers              = model.tracers
+    auxiliary_fields = model.auxiliary_fields
```
Comment (navidcy) [inline src/Models/NonhydrostaticModels/calculate_nonhydrostatic_tendencies.jl:52]:
```suggestion
    auxiliary_fields     = model.auxiliary_fields
```

File: docs/src/model_setup/forcing_functions.md [hunk 2]
```diff
@@ -97,7 +97,7 @@ of GPU-compiliability.
 
 ### Forcing functions that depend on model fields
 
-Forcing functions may depend on model fields evaluated at the `x, y, z` where forcing is applied.
+Forcing functions may depend on model fields (velocity, tracers or auxiliary fields) evaluated at the `x, y, z` where forcing is applied.
```
Comment (navidcy) [inline docs/src/model_setup/forcing_functions.md:100]: That's a nice addition. Does `ShallowWaterModel` accept auxiliary fields? Could we add that functionality there? Should we? cc @francispoulin

Comment (jagoosw) [reply docs/src/model_setup/forcing_functions.md:100]: Thanks. I've never used `ShallowWaterModels` myself so wanted to have a play with them before I tried, but it looks like it should be relatively straight forward like with the non hydrostatic model.

**Review by glwagner — APPROVED** [review]
I think this is great, and clean implementation!

One note is that this should probably be done uniformly across models as @navidcy says; basically nearly identical changes for `HydrostaticFreeSurfaceModel` in the constructor and definition of `fields`.

Possibly, the docs for `Forcing` should be updated as well.

Note that we have some explaining. Now `fields` is not _exactly_ prognostic or auxiliary fields, because some auxiliary fields are left out --- most notably, pressures and diffusivities. I think it's ok, we just need to say "field dependencies can be velocities, tracers, auxiliary fields, and free surface displacement".

File: docs/src/model_setup/forcing_functions.md (hunk 2 above)
Comment (jagoosw) [reply docs/src/model_setup/forcing_functions.md:100]: I've just started looking into implementing this and it looks like `auxiliary_fields` aren't implemented at all in `ShallowWaterModels`

**Comment by jagoosw** [comment]
Not sure why this test failed since I only changed some formatting

**Comment by navidcy** [comment]
Sometimes tests fail to load. Merely restarting them does the job. Did that. Let’s see.

**Comment by jagoosw** [comment]
Thank you

**Comment by navidcy** [comment]
@jagoosw I update the docstrings to mention auxiliary fields. I'll merge when tests pass. Thanks!

**Comment by jagoosw** [comment]
@navidcy fantastic, thank you!

**Pull request merged**